lib/state/analytics/middleware.ts

```diff
@@ -0,0 +1,82 @@
+import analytics from '../../analytics';
+import actions from '../actions';
+
+import * as A from '../action-types';
+import * as S from '../';
+import * as T from '../../types';
+
+export const middleware: S.Middleware = store => {
+  return next => (action: A.ActionType) => {
+    const result = next(action);
+    const nextState = store.getState();
+
+    // @todo events throughout the app that don't have a corresponding action
+    // analytics.tracks.recordEvent('user_account_created');
+    // analytics.tracks.recordEvent('editor_checklist_inserted');
+    // analytics.tracks.recordEvent('importer_import_completed', {
+    // analytics.tracks.recordEvent('editor_note_collaborator_added');
+    // analytics.tracks.recordEvent('editor_note_collaborator_removed');
+    // analytics.tracks.recordEvent('editor_note_edited');
+    // analytics.tracks.recordEvent('list_tag_deleted');
+
+    switch (action.type) {
+      case 'AUTH_SET':
+        if (action.status === 'authorized') {
+          // @todo called twice for some reason
```

Comment:
which is called twice? `AUTH_SET`, or `AUTH_SET` _with_ `status === 'authorized'`? because `AUTH_SET` should be called multiple times in succession throughout the sign-in process

Reply:
huh it WAS being called twice with status authorized, but maybe no longer..

Reply:
we should probably find out why if it's being called with with `authorized` - that wouldn't likely have been changed in years